aes/src/lib.rs

```diff
@@ -68,7 +68,17 @@ cfg_if! {
         any(target_arch = "x86_64", target_arch = "x86"),
     ))] {
         mod ni;
-        pub use ni::{Aes128, Aes192, Aes256};
+
+        cfg_if! {
```

Comment:
I think we should join all `cfg_if`s into a single one. The `union`-based proposal will make checking of `target_feature`s here redundant (i.e. with enabled target features the software fallback will be removed from the `autodetect` code making it equivalent to the `ni` implementation).

Reply:
Done